packages/user/src/User/Login.purs

```diff
@@ -198,38 +199,61 @@ failOnEmailMismatch self email
       liftEffect $ self.setState _ { errors { login = Just LoginEmailMismatchError } }
       throwError $ error "Emails don't match"
 
+loginWithRetry :: (Aff (Either UserError User)) -> (User -> Aff Unit) -> (UserError -> Aff Unit) -> (Either UserError User -> Aff Unit) -> Aff Unit
+loginWithRetry action handleSuccess handleError handleFinish = withRetry false
+  where
+    handleError' err = do
+      handleError err
+      pure true
+    withRetry retry = do
+      user <- action
+      notRetry <- case user of
+        Right user' -> do
+          handleSuccess user'
+          pure true
+        Left ServiceUnavailable -> do
+          if retry then handleError' ServiceUnavailable else do
```

Comment:
So if `retry = true` we don't actually retry? 😁 My brain is having hard time accepting this 😂

Reply:
I thought about it the way that `retry = true` if we are already doing the retry.

I was also considering using an integer but went with a boolean since I used only one retry. Perhaps it could be clearer to use it after all.